utam-compiler/src/main/java/utam/compiler/translator/ClassSerializer.java

```diff
@@ -94,22 +95,38 @@ public String toString() {
     out.add(NEW_LINE);
     source.getMethods().stream()
             // if method is private and never used, do not not declare to avoid test coverage alert
-            .filter(method -> isUsedMethod(method))
+            .filter(this::isUsedMethod)
             .flatMap(method -> getMethodDeclaration(method).stream()).forEach(out::add);
     out.add(NEW_LINE);
-    addElementClassDeclarations(out);
+    addPublicElementClassDeclarations(out);
+    addPrivateElementClassDeclarations(out);
     out.add(NEW_LINE);
     out.add("}");
     return applyJavaFormatter(out);
   }
 
-  private void addElementClassDeclarations(List<String> out) {
-    source.getMethods().stream()
-        .filter(method -> isUsedMethod(method) && method.isElementMethod())
-        .map(method -> method.getDeclaration().getReturnType())
-        .map(returnType -> returnType instanceof TypeUtilities.ListOf ?
-            ((TypeUtilities.ListOf)returnType).getElementType() :
-            returnType)
+  private void addPublicElementClassDeclarations(List<String> out) {
+    source.getDeclaredElementTypes(true).stream()
+        .map(returnType -> String.format(
+            "public static class %s extends %s implements %s {}",
+            returnType.getSimpleName() + "Impl",
+            BasePageElement.class.getSimpleName(),
+            returnType.getSimpleName()))
+        .forEach(out::add);
+  }
+
+  private void addPrivateElementClassDeclarations(List<String> out) {
+    List<TypeProvider> privateElementTypes = source.getDeclaredElementTypes(false);
+    privateElementTypes.stream()
+        .map(returnType -> String.format(
+            "public interface %s extends %s {}",
```

Comment:
i think "public" before interface is redundant, minor

Reply:
Of course, you're correct. Fixing in next commit.